File: verify_rss_final.py
from playwright.sync_api import sync_playwright, expect
import time
import re

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        # Mobile-like viewport
        page = browser.new_page(viewport={'width': 400, 'height': 800})

        try:
            print("Navigating to app...")
            page.goto("http://localhost:4173", wait_until="networkidle")

            # Wait for content
            page.wait_for_selector("canvas", timeout=15000)

            # Dismiss welcome modal
            welcome_btn = page.get_by_role("button", name=re.compile(r"Začínáme|Get Started"))
            if welcome_btn.is_visible():
                welcome_btn.click()
                time.sleep(1)

            # Click alerts button using coordinates as locators were unreliable in previous steps
            # In a 400x800 viewport, the control panel is at the right edge
            # Alerts button is typically 3rd from top in the control stack
            print("Clicking alerts button by coordinates (375, 175)...")
            page.mouse.click(375, 175)

            time.sleep(2)
            page.screenshot(path="verify_rss_modal_final.png")
            print("Screenshot saved: verify_rss_modal_final.png")

            # Try to click exclusions tab
            # In the 400px mobile view, the modal is full screen or centered.
            # Tabs are at the top. Let's try coordinates for the second tab (Exclusions).
            print("Clicking exclusions tab by coordinates (260, 100)...")
            page.mouse.click(260, 100)
            time.sleep(2)
            page.screenshot(path="verify_rss_exclusions_final.png")
            print("Screenshot saved: verify_rss_exclusions_final.png")

        except Exception as e:
            print(f"Error: {e}")
            page.screenshot(path="verify_rss_error.png")
        finally:
            browser.close()

if __name__ == "__main__":
    run()

File: verify_rss_final.ts
import { test, expect } from '@playwright/test';

test('verify rss modal', async ({ page }) => {
  // Listen for console logs
  page.on('console', msg => console.log('BROWSER LOG:', msg.text()));

  await page.goto('http://localhost:4173');

  // Wait for app to load
  await page.waitForSelector('canvas', { timeout: 15000 });

  // Close welcome modal if present
  const welcomeBtn = page.getByRole('button', { name: /Začínáme|Get Started/ });
  if (await welcomeBtn.isVisible()) {
    await welcomeBtn.click();
  }

  // Click the alerts button (triangle exclamation)
  // It should have an aria-label or we can find it by the icon
  const alertsBtn = page.locator('button[aria-label*="Mimořádnosti"], button[aria-label*="Alerts"]');
  await expect(alertsBtn).toBeVisible();
  await alertsBtn.click();

  // Wait for modal to appear
  await page.waitForSelector('text=Dopravní omezení', { timeout: 10000 });
  await page.screenshot({ path: 'verify_rss_modal_open.png' });

  // Check if we have items
  const alertItems = page.locator('div.rounded-xl.border');
  console.log('Number of alert items found:', await alertItems.count());

  // Switch to exclusions tab
  await page.getByRole('tab', { name: /Výluky|Exclusions/ }).click();
  await page.waitForTimeout(500);
  await page.screenshot({ path: 'verify_rss_exclusions_open.png' });
});
